rdflib/plugins/sparql/processor.py

```diff
@@ -63,8 +64,9 @@ def update(self, strOrQuery, initBindings={}, initNs={}):
 
 
 class SPARQLProcessor(Processor):
-    def __init__(self, graph):
+    def __init__(self, graph, optimizers: List[SPARQLOptimizer] = None):
```

Comment:
For simplicity sake I think it would be best to just accept any callable that maps a query to a query:

So somewhere before this:

```python
_QueryTranslatorType = Callable[[Query],Query]
```

Then:

```suggestion
    def __init__(self, graph, query_translators: Optional[List[_QueryTranslatorType]] = None):
```

Reply:
Your tests should still work fine with that.